pbcommand/models/common.py

```diff
@@ -527,3 +527,66 @@ def chunk_metadata(self):
 
     def to_dict(self):
         return {'chunk_id': self.chunk_id, 'chunk': self._datum}
+
+
+class DataStoreViewRule(object):
+    """
+    Rule specifying if and how the UI should display a datastore file.
+    """
+    def __init__(self, source_id, file_type_id, is_hidden, name=None,
+                 description=None):
+        # for generating rules compositionally in Python, it's easier to just
+        # pass the FileType object directly
+        if isinstance(file_type_id, FileType):
+            file_type_id = file_type_id.file_type_id
+        assert FileTypes.is_valid_id(file_type_id), file_type_id
+        self.source_id = source_id
+        self.file_type_id = file_type_id
+        self.is_hidden = is_hidden
+        self.name = name
+        self.description = description
+
+    def to_dict(self):
+        return {"id": self.source_id, "fileTypeId": self.file_type_id,
+                "isHidden": self.is_hidden, "name": self.name,
+                "description": self.description}
+
+    @staticmethod
+    def from_dict(d):
+        return DataStoreViewRule(d['id'], d['fileTypeId'], d['isHidden'],
+                                 d['name'], d['description'])
+
+
+class PipelineDataStoreViewRules(object):
+    """
+    A collection of DataStoreViewRule objects associated with a pipeline.
+    """
+
+    def __init__(self, pipeline_id, smrtlink_version, rules=()):
+        self.pipeline_id = pipeline_id
+        self.smrtlink_version = smrtlink_version
+        self.rules = list(rules)
+
+    def to_dict(self):
+        return {"pipelineId": self.pipeline_id,
+                "smrtlinkVersion": self.smrtlink_version,
+                "rules": [r.to_dict() for r in self.rules]}
+
+    @staticmethod
+    def from_dict(d):
+        return PipelineDataStoreViewRules(
+            pipeline_id=d['pipelineId'],
+            smrtlink_version=d['smrtlinkVersion'],
+            rules=[DataStoreViewRule.from_dict(r) for r in d['rules']])
+
+    @staticmethod
+    def load_from_json(path):
+        with open(path, 'r') as reader:
+            d = json.loads(reader.read())
+        return PipelineDataStoreViewRules.from_dict(d)
+
+    def write_json(self, file_name):
+        with open(file_name, "w") as f:
```

Comment:
Can  you push this core function to a centralize that has these custom json writing values?